File: pset2/caesar.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>

string cipher(string, int);

int main(int argc, string argv[]){
    
    if(argc != 2){
        printf("INVALID STATEMENT\n");
        return 1;
    }
    
	int key = (int)strtol(argv[1], (char **)NULL, 10); //ASCII adjustment
	
	printf("plaintext: ");
	string message = get_string();

	printf("ciphertext: %s\n", cipher(message, key) ); //printing the encrypted message
	return 0;
}

string cipher(string message, int key){
    
	// int length = strlen(message);

	for(int i = 0; message[i] != '\0'; i++){
	
	    if( isalpha(message[i]) ){
            
            if( isupper( message[i] ) ){
            	message[i] += key; //adding key
    			while( !isalpha( message[i] )) //if exceeding ASCII, go round // for UPPER
    				message[i] -= 26;
            }
            
			else if( islower( message[i] ) ){
				message[i] += key; //adding key
        		while( !isalpha( message[i] )) //if exceeding ASCII, go round // for UPPER
    				message[i] -= 26;
			}
		}
	}
	
	return message;
}

File: pset6/caesar/caesar.py
import sys

def cipher(message, key):
	message = list(message)
	cryp = ""
	for i in range(len(message)):
		ascii = ord(message[i])
		if message[i].isalpha():
			if message[i].isupper():
				ascii += key #adding key
				while ascii > 90: #if exceeding ASCII, go round # for UPPER
					ascii -= 26
				
			elif message[i].islower():
				ascii += key #adding key
				while ascii > 122: #if exceeding ASCII, go round # for LOWER
					ascii -= 26
		cryp += chr(ascii)
	return cryp


if (len(sys.argv) != 2):
	print("Usage: ./caesar k")
	sys.exit()
	
key = int(sys.argv[1])

message = input("plaintext: ")

print("ciphertext: %s" % cipher(message, key))